src/misc/fs_utils_posix.cpp

```diff
@@ -0,0 +1,98 @@
+/*
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *
+ *  Copyright (C) 2019-2020  The dosbox-staging team
+ *
+ *  This program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License along
+ *  with this program; if not, write to the Free Software Foundation, Inc.,
+ *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+
+#if !defined(WIN32)
+
+#include "fs_utils.h"
+
+#include <cctype>
+#include <glob.h>
+#include <unistd.h>
+
+#include "logging.h"
+#include "support.h"
+
+bool path_exists(const char *path) noexcept
+{
+	return (access(path, F_OK) == 0);
```

Comment:
Looking at the APIs of the Posix [`access()`](https://pubs.opengroup.org/onlinepubs/009695399/functions/access.html) and Windows [`_access()`](https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/access-waccess?view=vs-2019) functions versus what we're replacing (typically `stat(..) == 0`) calls - I don't see anything wrong.

We're also comparing the good-scenarios `*access(..) == 0` vs good stat scenario `stat(..) == 0`.  I could see us having issues if we replaced a not-failure check, such as `stat(..) != -1`, but we're not doing that.

Maybe `_access` is a bit more fragile than `stat`; so `access(..) != -1` might be more equal `stat(..) == 0`? 

Many stack-exchange answers seem to recommend the former:
- https://stackoverflow.com/a/12774387
- https://stackoverflow.com/a/46292862
- https://stackoverflow.com/a/230068

Some users commented that access() is still not as robust:
-  _"I ran across this question when looking for the reason access() broke in my code. I moved from DevC++ to CodeBlocks and it stopped working. "_
- _"Update: Note that on Windows, you can't use W_OK to reliably test for write permission, since the access function does not take DACLs into account. access( fname, W_OK ) may return 0 (success) because the file does not have the read-only attribute set, but you still may not have permission to write to the file."_
- _"There are pitfalls associated with access(). There is a TOCTOU (time of check, time of use) window of vulnerability between using access() and whatever else you do afterwards. [...to be continued...]"_

Maybe we could try swapping in a `stat()` call instead? 

Looking forward to C++17 :-) `std::filesystem::exists("path/to/file");`

Reply:
C++17 version is not good either… it communicates errors by throwing an exception :/ noexception version is `std::filesystem::exists("path/to/file", errorcode)` :/

Regarding W_OK - those do not work on Windows at all, that's why we use parameter 0 for `_access` function.
Eh, perhaps it's just safer to replace this with stat after all (I hoped access would be faster, because there's no big struct allocated on stack and function does not fill it with any data).

Tomorrow I'll test with 2 bugs from my commit addressed and investigate why different code path happens on Linux.